src/Microsoft.AspNetCore.Routing.Abstractions/RouteValueDictionary.cs

```diff
@@ -670,6 +669,104 @@ public override void Upgrade(ref Storage storage)
             }
         }
 
+        internal class KeyValuePairList
```

Comment:
This shouldn't be here, just put the code you need inside `ListStorage`. This will save a pointer, which is significant in a lot of our usage of this